cmd/headscale/cli/utils.go

```diff
@@ -40,6 +40,7 @@ func LoadConfig(path string) error {
 
 	viper.SetDefault("tls_letsencrypt_cache_dir", "/var/www/.cache")
 	viper.SetDefault("tls_letsencrypt_challenge_type", "HTTP-01")
+	viper.SetDefault("tls_client_auth_mode", "disabled")
```

Comment:
This default is different than the current setting, how come?

Reply:
Please correct me if I'm wrong, but I think Tailscale doesn't support client certificates right now.

The current default, [RequireAnyClientCert](https://cs.opensource.google/go/go/+/refs/tags/go1.17.6:src/crypto/tls/common.go;l=302), prevents connections over TLS unless any cert is provided by the client.

If I'm not overlooking something, then this means that Tailscale can't interact with Headscale when it's configured to use TLS.

Reply:
> If I'm not overlooking something, then this means that Tailscale can't interact with Headscale when it's configured to use TLS.

Hmm, I am not sure what this means? I am running my `headscale` with TLS?

Reply:
How curious! All of my browsers throw a bad ssl error unless a client certificate is supplied, then it works as expected. 

I'm using the config parameters to point at cert values for letsencrypt files I already have on hand.

I believe issue #254 reflects the same outcome. 

Do we have a misconfiguration somewhere or something?

Reply:
Hmm, happy to make it configurable so users with this problem can at least test them, however, there are many successful users (me included) and I think think there is a reason to change the default for everyone.

My current setup has tls set in nginx, but a week ago, I used the letsencrypt built in to headscale.

So yep, let's get it in, but, we should not change the default.

Reply:
Agreed! Sorry for the confusion there.

Should have things buttoned up tonight/early tomorrow.

Reply:
Alrighty, the latest commit preserves the original default.